vortex-array/src/array/chunked/mod.rs

```diff
@@ -132,6 +133,72 @@ impl ChunkedArray {
     pub fn array_stream(&self) -> impl ArrayStream + '_ {
         ArrayStreamAdapter::new(self.dtype().clone(), stream::iter(self.chunks().map(Ok)))
     }
+
+    pub fn rechunk(&self, target_bytesize: usize, target_rowsize: usize) -> VortexResult<Self> {
+        fn combine_validities(
+            dtype: &DType,
+            validities: Vec<LogicalValidity>,
+        ) -> VortexResult<Validity> {
+            let validity = validities.clone().into_iter().collect::<Validity>();
+            match (dtype.is_nullable(), validity) {
+                (true, validity) => Ok(validity),
+                (false, Validity::AllValid) => Ok(Validity::NonNullable),
+                (false, _) => vortex_bail!(
+                    "for non-nullable dtype, child validities ought to all be AllValid"
+                ),
+            }
```

Comment:
```suggestion
                  if self.dtype().is_nullable() {
                        validities_to_combine.iter().cloned().collect::<Validity>()
                    } else {
                        NonNullable
                    }
```
This can be simplified

Reply:
I killed combine_validity entirely by using one of your earlier suggestions to go through ChunkedArray. Does that seem reasonable?